hasher-matcher-actioner/terraform/stream-processor/main.tf

```diff
@@ -0,0 +1,107 @@
+
+# Define lambda
+resource "aws_lambda_function" "stream_processor" {
+  function_name = "${var.prefix}_stream_processor"
+  package_type  = "Image"
+  role          = aws_iam_role.stream_processor_lambda_role.arn
+  image_uri     = var.lambda_docker_info.uri
+  image_config {
+    command = [var.lambda_docker_info.commands.stream_processor]
+  }
+  timeout     = 300
+  memory_size = 512
+  environment {
+    variables = {
+      DYNAMODB_TABLE                        = var.datastore.name
+      HMA_CONFIG_TABLE                      = var.config_table.name
+      MEASURE_PERFORMANCE                   = var.measure_performance ? "True" : "False"
+    }
+  }
+  tags = merge(
+    var.additional_tags,
+    {
+      Name = "StreamProcessor"
+    }
+  )
+}
+
+resource "aws_cloudwatch_log_group" "stream_processor" {
+  name              = "/aws/lambda/${aws_lambda_function.stream_processor.function_name}"
+  retention_in_days = var.log_retention_in_days
+  tags = merge(
+    var.additional_tags,
+    {
+      Name = "StreamProcessorLambdaLogGroup"
+    }
+  )
+}
+
+# Lambda permisions, this goes on for a while.
+data "aws_iam_policy_document" "lambda_assume_role" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "stream_processor_lambda_role" {
+  name_prefix        = "${var.prefix}_stream_processor"
+  assume_role_policy = data.aws_iam_policy_document.lambda_assume_role.json
+  tags = merge(
+    var.additional_tags,
+    {
+      Name = "StreamProcessorLambdaRole"
+    }
+  )
+}
+
+data "aws_iam_policy_document" "stream_processor_iam_policy_document" {
+  statement {
+    effect    = "Allow"
+    actions   = ["dynamodb:GetItem", "dynamodb:Query", "dynamodb:Scan", "dynamodb:UpdateItem"]
+    resources = ["${var.datastore.arn}*"]
+  }
+  statement {
+    effect    = "Allow"
+    actions   = ["dynamodb:GetItem", "dynamodb:Query", "dynamodb:Scan", "dynamodb:PutItem", "dynamodb:DeleteItem"]
```

Comment:
[Diff 5/5] feels like an over Allow, are all these actions being used? if not can we wait to add them until they are?

Reply:
Evidently all of these are necessary to be able to read ddb streams.

Reply:
Nope. Copy-pasted. I'll limit it to the read only ones.